src/Data/Formatter/DateTime.purs

```diff
@@ -0,0 +1,426 @@
+           <*> (maybe (Left "incorrect month") pure $ toEnum $ fromMaybe one a.month)
+           <*> (maybe (Left "incorrect day") pure $ toEnum $ fromMaybe one a.day))
+    <*> (T.Time
+           <$> (maybe
+                  (Left "incorrect hour") pure
+                  $ toEnum
+                  $ (if a.meridiem == Just PM then (_ + 12) else id)
+                  $ fromMaybe zero a.hour)
+           <*> (maybe (Left "incorrect minute") pure $ toEnum $ fromMaybe zero a.minute)
+           <*> (maybe (Left "incorrect second") pure $ toEnum $ fromMaybe zero a.second)
+           <*> (maybe (Left "incorrect millisecond") pure $ toEnum $ fromMaybe zero a.millisecond))
+
+
+unformatFParser
+  ∷ ∀ a
+  . (a → P.ParserT String (State UnformatAccum) Unit)
+  → FormatterF a
+  → P.ParserT String (State UnformatAccum) Unit
+unformatFParser cb = case _ of
+  YearFull a → do
+    ds ← some digit
+    when (Arr.length ds /= 4) $ P.fail "Incorrect full year"
+    lift $ modify _{year = Just $ foldDigits ds}
+    cb a
+  YearTwoDigits a → do
+    ds ← some digit
+    when (Arr.length ds /= 2) $ P.fail "Incorrect 2digit year"
+    lift $ modify _{year = Just $ foldDigits ds + 1900}
+    cb a
+  YearAbsolute a → do
+    sign ← PC.optionMaybe $ PC.try $ PS.string "-"
+    year ← map foldDigits $ some digit
+    lift $ modify _{year = Just $ (if isJust sign then -1 else 1) * year}
+    cb a
+  MonthFull a → do
+    month ← parseMonth
+    lift $ modify _{month = Just $ fromEnum month}
+    cb a
+  MonthShort a → do
+    month ← parseShortMonth
+    lift $ modify _{month = Just $ fromEnum month}
+    cb a
+  MonthTwoDigits a → do
+    ds ← some digit
+    let month = foldDigits ds
+    when (Arr.length ds /= 2 || month > 12 || month < 1) $ P.fail "Incorrect 2digit month"
+    lift $ modify _{month = Just month}
+    cb a
+  DayOfMonth a → do
+    ds ← some digit
+    let dom = foldDigits ds
+    when (Arr.length ds /= 2 || dom > 31 || dom < 1) $ P.fail "Incorrect day of month"
+    lift $ modify _{day = Just dom}
+    cb a
+  UnixTimestamp a → do
+    s ← map foldDigits $ some digit
+    case map toDateTime $ instant $ Dur.Milliseconds $ 1000.0 * Int.toNumber s of
+      Nothing → P.fail "Incorrect timestamp"
+      Just (DT.DateTime d t) → do
+        lift $ put { year: Just $ fromEnum $ D.year d
+                   , month: Just $ fromEnum $ D.month d
+                   , day: Just $ fromEnum $ D.day d
+                   , hour: Just $ fromEnum $ T.hour t
+                   , minute: Just $ fromEnum $ T.minute t
+                   , second: Just $ fromEnum $ T.second t
+                   , millisecond: Just $ fromEnum $ T.millisecond t
+                   , meridiem: (Nothing ∷ Maybe Meridiem)
+                   , dayOfWeek: (Nothing ∷ Maybe Int)
+                   }
+        cb a
+  DayOfWeek a → do
+    dow ← digit
+    when (dow > 7 || dow < 1) $ P.fail "Incorrect day of week"
+    lift $ modify _{dayOfWeek = Just dow}
+    cb a

```

Comment:
This is a strange one in the unformat situation - do we need to store it at all? We basically ignore it when constructing the date, so keeping the info in the accum is unnecessary perhaps.